mmpose/codecs/udp_heatmap.py

```diff
@@ -97,16 +97,30 @@ def encode(
             - keypoint_weights (np.ndarray): The target weights in shape
                 (K,)
         """
+        assert keypoints.shape[0] == 1, (
+            f'{self.__class__.__name__} only support single-instance '
+            'keypoint encoding')
+
+        if keypoints_visible is None:
+            keypoints_visible = np.ones(keypoints.shape[:2], dtype=np.float32)
+
         if self.heatmap_type == 'gaussian':
-            return self._encode_gaussian(keypoints, keypoints_visible)
+            heatmaps, keypoint_weights = generate_udp_gaussian_heatmaps(
+                heatmap_size=self.heatmap_size,
+                keypoints=keypoints / self.scale_factor,
+                keypoints_visible=keypoints_visible,
+                sigma=self.sigma)
         elif self.heatmap_type == 'combined':
-            return self._encode_combined(keypoints, keypoints_visible)
+            heatmaps, keypoint_weights = self._encode_combined(
```

Comment:
I suggest moving `encode_combined` to utils/gaussian_heatmap.py as well. It can be commonly used  by other offset-based methods

Reply:
Agree. As for the function name, what about `generate_offset_map`? Or we break it into 2 functions `generate_binary_heatmap` and `generate_offset_map`.

Reply:
What about `generate_offset_heatmap`? I think it's a fixed paradigm, which is complete enough. Anyone wants to change this paradigm should implement a new function.